File: Placement Sheet/Stack & Queue/BS/MinStack.java
import java.util.Stack;

public class MinStack {
    Stack<Integer> stack = new Stack<>();
    Stack<Integer> minStack = new Stack<>();

    // Push...
    public void push(int data) {
        stack.push(data);
        if (minStack.size() == 0 || data <= getMin()) {
            minStack.push(data);
        }
    }

    // Pop...
    public int pop() {
        int poppedElm = stack.pop();
        if (getMin() == poppedElm) {
            minStack.pop();
        }
        return poppedElm;
    }

    // Peek...
    public int peek() {
        return stack.peek();
    }

    public int getMin() {
        return minStack.peek();
    }

    public static void main(String[] args) {
        MinStack stack = new MinStack();
        stack.push(5);
        stack.push(3);
        stack.push(7);
        stack.push(2);

        System.out.println("Min: " + stack.getMin());

        stack.pop();
        System.out.println("Min: " + stack.getMin());
        System.out.println("Top: " + stack.peek());
    }
}

File: Placement Sheet/Stack & Queue/BS/MinStack.ts
class MinStack<T> {
    private stack: T[] = [];
    private minStack: T[] = [];

    //Push...
    push(data: T): void {
        this.stack.push(data);
        if (this.minStack.length == 0 || data <= this.getMin()) {
            this.minStack.push(data);
        }
    }

    //Pop...
    pop(): T | undefined {
        const poppedElm: T | undefined = this.stack.pop();
        if (poppedElm === this.getMin()) {
            this.minStack.pop();
        }
        return poppedElm;
    }

    //Peek...
    peek(): T | undefined {
        return this.stack[this.stack.length - 1];
    }

    getMin(): T {
        return this.minStack[this.minStack.length - 1];
    }
}

let minStack: MinStack<number> = new MinStack<number>();

minStack.push(5);
minStack.push(3);
minStack.push(7);
minStack.push(2);

console.log(`Min: ${minStack.getMin()}`);

minStack.pop();
console.log(`Min: ${minStack.getMin()}`);
console.log(`Top: ${minStack.peek()}`);

export { };